# Sentiment annotation of 18th Century Spanish periodicals


## Introduction

In this Jupyter Notebook we evaluate the annotations of 50 senteces extracted from Spanish 18th century periodicals, building on the Dispecs project. The annotations were produced by two expert annotators. The annotation study was conducted through a survey, created on the Lime Survey site and the data was extracted in form of comma-delimited .csv files. The annotators could choose between the following polarity categories: positive, negative and neutral as well as the following emotion categories: joy, trust, anticipation, surprise, anger, disgust, sadness, and fear. To be able to run this notebook, you'll need to import the libraries pandas, nltk and the agreement package from nltk, which will allow us to calculate the inter-rater agreements.

In [1]:
import pandas as pd
import nltk
from nltk import agreement

In the following .txt and information dataframe, you can find information about the 50 annotated sentences : which question in the survey they correspond, which narrative form they were taken from as well as what is the name of periodical from which they were taken. The cells marked as '??' signify sentences that still need to be associated with a narrative form.

In [2]:
information = pd.read_csv('input/question_nf_periodical.txt', delimiter=',', encoding='utf-8')
information.columns = ['question_code', 'narrative_form','periodical_name']
information

question_code narrative_form  \
0             Q2             ??   
1             Q3             ??   
2             Q4             ??   
3             Q5             ??   
4             Q6             ??   
5            Q50              D   
6            Q51             FP   
7             Q7             ??   
8             Q8             ??   
9            Q52              D   
10           Q53             SP   
11            Q9             LB   
12           Q54             EX   
13           Q10             FP   
14           Q55             EX   
15           Q11             FP   
16           Q12             ??   
17           Q13             MT   
18           Q14             SP   
19           Q56              S   
20           Q57              S   
21           Q15              D   
22           Q16              D   
23           Q58             EX   
24           Q59             FP   
25           Q60              S   
26           Q17             EX   
27           Q61             TR   
28           Q62             TR   
29           Q18             FP   
30           Q63             MT   
31           Q64             LB   
32           Q65             LB   
33          Q119             FP   
34           Q20             SP   
35           Q66             LB   
36           Q21             ??   
37           Q67             MT   
38           Q68             TR   
39           Q69             MT   
40            70             LB   
41           Q71             FP   
42           Q23             FP   
43           Q24             LB   
44           Q25             FP   
45           Q26             FP   
46           Q27              F   
47           Q28              S   

                                      periodical_name  
0   1761-07-08_El-Duende-especulativo-sobre-la-vid...  
1   1761-07-27_El-Duende-especulativo-sobre-la-vid...  
2   1761-09-05_El-Duende-especulativo-sobre-la-vid...  
3    1761-09-26_El-Duende-especulativo-sobre-la-vi...  
4   1761-07-27_El-Duende-especulativo-sobre-la-vid...  
5    1786_El-Juzgado-Casero_Anónimo_Vol-1_Nr-01_47...  
6    1786_El-Juzgado-Casero_Anónimo_Vol-1_Nr-01_47...  
7    1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
8   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallada...  
9    1787-1788_El-Duende-de-Madrid_Pedro-Pablo-Tru...  
10   1787_El-Censor_Anonym-(García-de-Cañuelo-Luis...  
11   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
12   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
13   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
14   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
15  1762-1763_El-Pensador_Joseph-Álvarez-y-Vallada...  
16   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
17   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
18   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
19   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
20   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
21   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
22   1762-1763_El-Pensador_Joseph-Álvarez-y-Vallad...  
23   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
24   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
25   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-1_Nr-0...  
26   1763_La-Pensadora-Gaditana_Beatriz-Cienfuegos...  
27   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-2_Nr-0...  
28   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-2_Nr-0...  
29   1763_La-Pensadora-Gaditana_Beatriz-Cienfuegos...  
30   1788_El-Filósofo-á-la-Moda_Anónimo_Vol-2_Nr-0...  
31   1800_El-Catón-Compostelano_Anónimo-(Francisco...  
32   1803_El-Regañón-general_Anónimo-(Ventura-Ferr...  
33   67_El-Pensador_Joseph-Álvarez-y-Valladares-(J...  
34   67_El-Pensador_Joseph-Álvarez-y-Valladares-(J...  
35   1803_El-Regañón-general_Anónimo-(Ventura-Ferr...  
36   1779-1780_El-Curioso-Entretenido_Juan-Nosip-y...  
37   1803_El-Regañón-general_Anónimo-(Ventura-Ferr...  
38   1803_El-Regañón-general_Anónimo-(Ventura-Ferr...  
39   1803_El-Regañón-general_Anónimo-(Ventura-F

## 1. Study with two annotators (Round 1)

### 1.1 Agreement over polarity and emotion annotations combined


The following are the results of inter-rater agreement between two expert annotators on all categories (both polarities and emotions). We're importing the survey results in a .csv format, where the answer yes was transformed into 1 and no into 0. 

In [3]:
survey_results = pd.read_csv('input/1_round/1_full_survey_polarities_emotions.csv', delimiter=',', encoding='utf-8')

In [4]:
survey_results

id  G01Q01[1]  G01Q01[2]  G01Q01[3]  G01Q01[5]  G01Q01[12]  G01Q01[6]  \
0   1          0          1          0          0           0          0   
1   2          0          0          1          0           0          0   

   G01Q01[10]  G01Q01[7]  G01Q01[8]  ...  G01Q28[2]  G01Q28[3]  G01Q28[5]  \
0           0          0          0  ...          1          0          0   
1           0          1          0  ...          0          1          0   

   G01Q28[12]  G01Q28[6]  G01Q28[10]  G01Q28[7]  G01Q28[8]  G01Q28[9]  \
0           0          0           0          0          0          0   
1           0          0           0          0          0          0   

   G01Q28[11]  
0           0  
1           0  

[2 rows x 551 columns]

The NLTK.metrics package which we will use to calculate the agreement demands the data argument to be in a sequence of 3-tuples, each representing a coder's labeling of theitem (coder, item, label) (see:https://www.nltk.org/api/nltk.metrics.agreement.html). Therefore, the first step to transform our data is to transpose the dataframe, after which some changes aremade (dropping the first three rows which we didn't need for the agreement and renaming the columns to annotator ids). In this process we are creating a new dataframe called 'results'.

In [5]:
results=survey_results.T
results.drop(['id'], axis=0, inplace=True)
results.columns = ['id_1', 'id_2']

In [6]:
results


id_1  id_2
G01Q01[1]      0     0
G01Q01[2]      1     0
G01Q01[3]      0     1
G01Q01[5]      0     0
G01Q01[12]     0     0
...          ...   ...
G01Q28[10]     0     0
G01Q28[7]      0     0
G01Q28[8]      0     0
G01Q28[9]      0     0
G01Q28[11]     0     0

[550 rows x 2 columns]

In this step we are creating the 3-tuples from our dataframe. We first create a list, and then iterate over indices and rows of our dataframe (results) and append the following data to the list - annotator id, row index and the contents of each row (considering that the rows are in pairs, first one annotator rates one subject then the other).

In [7]:
data = []
for idx, row in results.iterrows():
    data.append(("id_1", idx, row["id_1"]))
    data.append(("id_2", idx, row["id_2"]))
data

[('id_1', 'G01Q01[1]', 0),
 ('id_2', 'G01Q01[1]', 0),
 ('id_1', 'G01Q01[2]', 1),
 ('id_2', 'G01Q01[2]', 0),
 ('id_1', 'G01Q01[3]', 0),
 ('id_2', 'G01Q01[3]', 1),
 ('id_1', 'G01Q01[5]', 0),
 ('id_2', 'G01Q01[5]', 0),
 ('id_1', 'G01Q01[12]', 0),
 ('id_2', 'G01Q01[12]', 0),
 ('id_1', 'G01Q01[6]', 0),
 ('id_2', 'G01Q01[6]', 0),
 ('id_1', 'G01Q01[10]', 0),
 ('id_2', 'G01Q01[10]', 0),
 ('id_1', 'G01Q01[7]', 0),
 ('id_2', 'G01Q01[7]', 1),
 ('id_1', 'G01Q01[8]', 0),
 ('id_2', 'G01Q01[8]', 0),
 ('id_1', 'G01Q01[9]', 0),
 ('id_2', 'G01Q01[9]', 0),
 ('id_1', 'G01Q01[11]', 0),
 ('id_2', 'G01Q01[11]', 0),
 ('id_1', 'G01Q02[1]', 0),
 ('id_2', 'G01Q02[1]', 0),
 ('id_1', 'G01Q02[2]', 0),
 ('id_2', 'G01Q02[2]', 0),
 ('id_1', 'G01Q02[3]', 1),
 ('id_2', 'G01Q02[3]', 1),
 ('id_1', 'G01Q02[5]', 0),
 ('id_2', 'G01Q02[5]', 0),
 ('id_1', 'G01Q02[12]', 0),
 ('id_2', 'G01Q02[12]', 0),
 ('id_1', 'G01Q02[6]', 0),
 ('id_2', 'G01Q02[6]', 0),
 ('id_1', 'G01Q02[10]', 0),
 ('id_2', 'G01Q02[10]', 0),
 ('id_1', 'G01Q02[

In this step we are calculating Cohen's kappa, Fleiss' Kappa, Krippendorf's Alpha and Scott's Pi for the whole set of annotations using nltk's agreement.AnnotationTask. The results point to the egreement being fair (0.32)

In [8]:
atask = agreement.AnnotationTask(data=data)
print("Cohen's Kappa:", atask.kappa())
print("Fleiss's Kappa:", atask.multi_kappa())
print("Krippendorf's Alpha:", atask.alpha())
print("Scott's Pi", atask.pi())

Cohen's Kappa: 0.31640387005580883
Fleiss's Kappa: 0.31640387005580883
Krippendorf's Alpha: 0.31695873650344875
Scott's Pi 0.3163372248897124


### 1.2 Polarity annotation agreement

In this section we are repeating the steps from the previous one, except that the agreement is calculated only for the polarity annotations (positive, negative, neutral)

In [9]:
polarities_survey = pd.read_csv('input/1_round/1_survey_polarities.csv', delimiter=',', encoding='utf-8')
polarities_results = polarities_survey.T
polarities_results.columns = ['id_1', 'id_2']
polarities_results

id_1  id_2
id            4     6
G01Q01[1]     0     0
G01Q01[2]     1     0
G01Q01[3]     0     1
G01Q02[1]     0     0
...         ...   ...
G01Q27[2]     1     1
G01Q27[3]     0     0
G01Q28[1]     0     0
G01Q28[2]     1     0
G01Q28[3]     0     1

[151 rows x 2 columns]

In [10]:
data_polarities = []
for idx, row in polarities_results.iterrows():
    data_polarities.append(("id_1", idx, row["id_1"]))
    data_polarities.append(("id_2", idx, row["id_2"]))
data_polarities

[('id_1', 'id', 4),
 ('id_2', 'id', 6),
 ('id_1', 'G01Q01[1]', 0),
 ('id_2', 'G01Q01[1]', 0),
 ('id_1', 'G01Q01[2]', 1),
 ('id_2', 'G01Q01[2]', 0),
 ('id_1', 'G01Q01[3]', 0),
 ('id_2', 'G01Q01[3]', 1),
 ('id_1', 'G01Q02[1]', 0),
 ('id_2', 'G01Q02[1]', 0),
 ('id_1', 'G01Q02[2]', 0),
 ('id_2', 'G01Q02[2]', 0),
 ('id_1', 'G01Q02[3]', 1),
 ('id_2', 'G01Q02[3]', 1),
 ('id_1', 'G01Q03[1]', 1),
 ('id_2', 'G01Q03[1]', 0),
 ('id_1', 'G01Q03[2]', 0),
 ('id_2', 'G01Q03[2]', 0),
 ('id_1', 'G01Q03[3]', 0),
 ('id_2', 'G01Q03[3]', 1),
 ('id_1', 'G01Q04[1]', 0),
 ('id_2', 'G01Q04[1]', 0),
 ('id_1', 'G01Q04[2]', 1),
 ('id_2', 'G01Q04[2]', 1),
 ('id_1', 'G01Q04[3]', 0),
 ('id_2', 'G01Q04[3]', 0),
 ('id_1', 'G01Q05[1]', 1),
 ('id_2', 'G01Q05[1]', 1),
 ('id_1', 'G01Q05[2]', 0),
 ('id_2', 'G01Q05[2]', 0),
 ('id_1', 'G01Q05[3]', 0),
 ('id_2', 'G01Q05[3]', 0),
 ('id_1', 'G01Q06[1]', 0),
 ('id_2', 'G01Q06[1]', 0),
 ('id_1', 'G01Q06[2]', 0),
 ('id_2', 'G01Q06[2]', 0),
 ('id_1', 'G01Q06[3]', 1),
 ('id_2', 'G01Q

In [11]:
atask_polarities = agreement.AnnotationTask(data=data_polarities)
print("Cohen's Kappa:", atask_polarities.kappa())
print("Fleiss's Kappa:", atask_polarities.multi_kappa())
print("Krippendorf's Alpha:", atask_polarities.alpha())
print("Scott's Pi", atask_polarities.pi())

Cohen's Kappa: 0.31103776332394883
Fleiss's Kappa: 0.31103776332394883
Krippendorf's Alpha: 0.31328576282704723
Scott's Pi 0.311004320178632


The polarity agreement is 0.31, which fits the category "fair agreement".

### 1.3 Emotion annotation agreement

In this section, we are repeating the previous steps for the emotion annotations. There were 7 categories in total, namely: joy, trust, anticipation, surprise, anger, disgust, sadness, and fear. 

In [12]:
emotions_survey = pd.read_csv('input/1_round/1_survey_emotions.csv', delimiter=',', encoding='utf-8')
emotions_results = emotions_survey.T
emotions_results.columns = ['id_1', 'id_2']
emotions_results

id_1  id_2
G01Q01[5]      0     0
G01Q01[12]     0     0
G01Q01[6]      0     0
G01Q01[10]     0     0
G01Q01[7]      0     1
...          ...   ...
G01Q28[10]     0     0
G01Q28[7]      0     0
G01Q28[8]      0     0
G01Q28[9]      0     0
G01Q28[11]     0     0

[400 rows x 2 columns]

In [13]:
data_emotions = []
for idx, row in emotions_results.iterrows():
    data_emotions.append(("id_1", idx, row["id_1"]))
    data_emotions.append(("id_2", idx, row["id_2"]))
data_emotions

[('id_1', 'G01Q01[5]', 0),
 ('id_2', 'G01Q01[5]', 0),
 ('id_1', 'G01Q01[12]', 0),
 ('id_2', 'G01Q01[12]', 0),
 ('id_1', 'G01Q01[6]', 0),
 ('id_2', 'G01Q01[6]', 0),
 ('id_1', 'G01Q01[10]', 0),
 ('id_2', 'G01Q01[10]', 0),
 ('id_1', 'G01Q01[7]', 0),
 ('id_2', 'G01Q01[7]', 1),
 ('id_1', 'G01Q01[8]', 0),
 ('id_2', 'G01Q01[8]', 0),
 ('id_1', 'G01Q01[9]', 0),
 ('id_2', 'G01Q01[9]', 0),
 ('id_1', 'G01Q01[11]', 0),
 ('id_2', 'G01Q01[11]', 0),
 ('id_1', 'G01Q02[5]', 0),
 ('id_2', 'G01Q02[5]', 0),
 ('id_1', 'G01Q02[12]', 0),
 ('id_2', 'G01Q02[12]', 0),
 ('id_1', 'G01Q02[6]', 0),
 ('id_2', 'G01Q02[6]', 0),
 ('id_1', 'G01Q02[10]', 0),
 ('id_2', 'G01Q02[10]', 0),
 ('id_1', 'G01Q02[7]', 0),
 ('id_2', 'G01Q02[7]', 0),
 ('id_1', 'G01Q02[8]', 0),
 ('id_2', 'G01Q02[8]', 0),
 ('id_1', 'G01Q02[9]', 0),
 ('id_2', 'G01Q02[9]', 1),
 ('id_1', 'G01Q02[11]', 1),
 ('id_2', 'G01Q02[11]', 0),
 ('id_1', 'G01Q03[5]', 0),
 ('id_2', 'G01Q03[5]', 0),
 ('id_1', 'G01Q03[12]', 0),
 ('id_2', 'G01Q03[12]', 0),
 ('id_1', 'G01

In [14]:
atask_emotions = agreement.AnnotationTask(data=data_emotions)
print("Cohen's Kappa:", atask_emotions.kappa())
print("Fleiss's Kappa:", atask_emotions.multi_kappa())
print("Krippendorf's Alpha:", atask_emotions.alpha())
print("Scott's Pi", atask_emotions.pi())

Cohen's Kappa: 0.20471490449478116
Fleiss's Kappa: 0.20471490449478116
Krippendorf's Alpha: 0.20548334398749912
Scott's Pi 0.20448895518147572


The emotion agreement is only 0.204, which is on the border between slight and fair agreement.

## 2. Study with three annotators (round 2)

In the following, a new annotation study was conducted with three annotators for both emotions (Eckman) and polarities(pos/neg/neu).

In [15]:
sentiment1 = pd.read_csv('input/2_round/part1_sentiment.csv', delimiter=',', encoding='utf-8')
sentiment2 = pd.read_csv('input/2_round/part2_sentiment.csv', delimiter=',', encoding='utf-8')
emotion1 = pd.read_csv('input/2_round/part1_emotion.csv', delimiter=',', encoding='utf-8')
emotion2 = pd.read_csv('input/2_round/part2_emotion.csv', delimiter=',', encoding='utf-8')

In [16]:
sentiment1

id  G01Q01[1]  G01Q01[2]  G01Q01[3]  G01Q02[1]  G01Q02[2]  G01Q02[3]  \
0   2          0          0          1          0          1          0   
1   1          0          0          1          0          1          0   
2   4          0          0          1          0          0          1   

   G01Q03[1]  G01Q03[2]  G01Q03[3]  ...  G00Q133[3]  G00Q134[1]  G00Q134[2]  \
0          0          0          1  ...           1           0           0   
1          0          0          1  ...           1           0           0   
2          0          0          1  ...           1           0           0   

   G00Q134[3]  G00Q135[1]  G00Q135[2]  G00Q135[3]  G00Q25[1]  G00Q25[2]  \
0           1           1           0           0          0          1   
1           1           0           0           1          0          0   
2           1           0           0           1          0          0   

   G00Q25[3]  
0          0  
1          1  
2          1  

[3 rows x 409 columns]

In [17]:
sentiment2

id  G00Q136[1]  G00Q136[2]  G00Q136[3]  G00Q137[1]  G00Q137[2]  G00Q137[3]  \
0   1           0           0           1           0           0           1   
1   2           0           0           1           0           0           1   
2   4           0           0           1           0           0           1   

   G00Q138[1]  G00Q138[2]  G00Q138[3]  ...  G00Q267[3]  G00Q268[1]  \
0           0           1           0  ...           1           1   
1           0           1           0  ...           1           1   
2           0           0           1  ...           1           0   

   G00Q268[2]  G00Q268[3]  G00Q269[1]  G00Q269[2]  G00Q269[3]  G00Q270[1]  \
0           0           0           0           1           0           1   
1           0           0           0           1           0           1   
2           0           1           0           0           1           1   

   G00Q270[2]  G00Q270[3]  
0           0           0  
1           0           0  
2           0           0  

[3 rows x 406 columns]

In [18]:
emotion1

id  \
0   2   
1   1   
2   4   

   G01Q01[5].  __Previous_sentence:__Pero_ella,_conociendo_el_riesgo_que_havia_en_verle,_y_escucharle,_llena_de_indignacion,_y_colera,_le_negò_su_presencia.__ __To_annotate: __Mi_pobre_Vecino,_posseìdo_de_una_furiosa_frenesì,_enagenado_de_sì_mismo,_y_sofocado_de_rabia,_se_enfurecia,_y_maldecia_su_fortuna,_y_precipitada_audácia.__ __Following_sentence: __Hasta_que_bolviendo_en_sì,_se_hallò_confuso_de_sus_extravagancias,_è_intentò,_arrepentido,_trocar_su_enojo_en_una_sumission_profunda;_pero_nada_sirviò_para_el_caso. __ __ __[joy]  \
0                                                  0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
1                                                  0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
2                                                  0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     

   G01Q01[12].  __Previous_sentence:__Pero_ella,_conociendo_el_riesgo_que_havia_en_verle,_y_escucharle,_llena_de_indignacion,_y_colera,_le_negò_su_presencia.__ __To_annotate: __Mi_pobre_Vecino,_posseìdo_de_una_furiosa_frenesì,_enagenado_de_sì_mismo,_y_sofocado_de_rabia,_se_enfurecia,_y_maldecia_su_fortuna,_y_precipitada_audácia.__ __Following_sentence: __Hasta_que_bolviendo_en_sì,_se_hallò_confuso_de_sus_extravagancias,_è_intentò,_arrepentido,_trocar_su_enojo_en_una_sumission_profunda;_pero_nada_sirviò_para_el_caso. __ __ __[trust]  \
0                                                  0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
1                                                  0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
2                                                  0                                                                                                                                          

In [19]:
emotion2

id  \
0   1   
1   2   
2   4   

   G00Q136[5]. Previous_sentence:__Diceles_palabras_injuriosas,_y_ellos_la_responden_sin_decoro.__To_annotate: __Miralos_sin_humanidad,_y_como_esclavos,_y_ellos_à_su_vez_la_sirven_sin_respeto,_sublevados_contra_su_tyranìa. __Following_sentence: __Suspendese_por_un_rato_esta_gresca,_y_passa_Madama_al_Tocador. __[joy]  \
0                                                  0                                                                                                                                                                                                                                                                            
1                                                  0                                                                                                                                                                                                                                                                            
2                                                  0                                                                                                                                                                                                                                                                            

   G00Q136[12]. Previous_sentence:__Diceles_palabras_injuriosas,_y_ellos_la_responden_sin_decoro.__To_annotate: __Miralos_sin_humanidad,_y_como_esclavos,_y_ellos_à_su_vez_la_sirven_sin_respeto,_sublevados_contra_su_tyranìa. __Following_sentence: __Suspendese_por_un_rato_esta_gresca,_y_passa_Madama_al_Tocador. __[trust]  \
0                                                  0                                                                                                                                                                                                                                                                               
1                                                  0                                                                                                                                                                                                                                                                               
2                                                  0                                                                                                                                                                                                                                                                               

   G00Q136[6]. Previous_sentence:__Diceles_palabras_injuriosas,_y_ellos_la_responden_sin_decoro.__To_annotate: __Miralos_sin_humanidad,_y_como_esclavos,_y_ellos_à_su_vez_la_sirven_sin_respeto,_sublevados_contra_su_tyranìa. __Following_sentence: __Suspendese_por_un_rato_esta_gresca,_y_passa_Madama_al_Tocador. __[anticipation]  \
0                                                  0                                                                                                                                                                                                                                                                                     
1                                                  0                                                                                                                                                                                                                                                                                     
2                                                  0                                                                                                                                                                                                                                                                                     

   G00Q136[10]. Previous_sentence:__Diceles_palabras_injuriosas

In [20]:
sentiment1_res = sentiment1.T
sentiment1_res.columns = ['id_2', 'id_1','id_4']
sentiment1_res = sentiment1_res.drop('id')
sentiment1_res

id_2  id_1  id_4
G01Q01[1]      0     0     0
G01Q01[2]      0     0     0
G01Q01[3]      1     1     1
G01Q02[1]      0     0     0
G01Q02[2]      1     1     0
...          ...   ...   ...
G00Q135[2]     0     0     0
G00Q135[3]     0     1     1
G00Q25[1]      0     0     0
G00Q25[2]      1     0     0
G00Q25[3]      0     1     1

[408 rows x 3 columns]

In [21]:
sentiment2_res = sentiment2.T
sentiment2_res.columns = ['id_1', 'id_2','id_4']
sentiment2_res = sentiment2_res.drop('id')
sentiment2_res

id_1  id_2  id_4
G00Q136[1]     0     0     0
G00Q136[2]     0     0     0
G00Q136[3]     1     1     1
G00Q137[1]     0     0     0
G00Q137[2]     0     0     0
...          ...   ...   ...
G00Q269[2]     1     1     0
G00Q269[3]     0     0     1
G00Q270[1]     1     1     1
G00Q270[2]     0     0     0
G00Q270[3]     0     0     0

[405 rows x 3 columns]

In [22]:
sentiment_res = pd.concat([sentiment1_res, sentiment2_res])
sentiment_res

id_2  id_1  id_4
G01Q01[1]      0     0     0
G01Q01[2]      0     0     0
G01Q01[3]      1     1     1
G01Q02[1]      0     0     0
G01Q02[2]      1     1     0
...          ...   ...   ...
G00Q269[2]     1     1     0
G00Q269[3]     0     0     1
G00Q270[1]     1     1     1
G00Q270[2]     0     0     0
G00Q270[3]     0     0     0

[813 rows x 3 columns]

In [23]:
data_sentiment = []
for idx, row in sentiment_res.iterrows():
    data_sentiment.append(("id_2", idx, row["id_2"]))
    data_sentiment.append(("id_1", idx, row["id_1"]))
    data_sentiment.append(("id_4", idx, row["id_4"]))
data_sentiment

[('id_2', 'G01Q01[1]', 0),
 ('id_1', 'G01Q01[1]', 0),
 ('id_4', 'G01Q01[1]', 0),
 ('id_2', 'G01Q01[2]', 0),
 ('id_1', 'G01Q01[2]', 0),
 ('id_4', 'G01Q01[2]', 0),
 ('id_2', 'G01Q01[3]', 1),
 ('id_1', 'G01Q01[3]', 1),
 ('id_4', 'G01Q01[3]', 1),
 ('id_2', 'G01Q02[1]', 0),
 ('id_1', 'G01Q02[1]', 0),
 ('id_4', 'G01Q02[1]', 0),
 ('id_2', 'G01Q02[2]', 1),
 ('id_1', 'G01Q02[2]', 1),
 ('id_4', 'G01Q02[2]', 0),
 ('id_2', 'G01Q02[3]', 0),
 ('id_1', 'G01Q02[3]', 0),
 ('id_4', 'G01Q02[3]', 1),
 ('id_2', 'G01Q03[1]', 0),
 ('id_1', 'G01Q03[1]', 0),
 ('id_4', 'G01Q03[1]', 0),
 ('id_2', 'G01Q03[2]', 0),
 ('id_1', 'G01Q03[2]', 0),
 ('id_4', 'G01Q03[2]', 0),
 ('id_2', 'G01Q03[3]', 1),
 ('id_1', 'G01Q03[3]', 1),
 ('id_4', 'G01Q03[3]', 1),
 ('id_2', 'G01Q04[1]', 0),
 ('id_1', 'G01Q04[1]', 0),
 ('id_4', 'G01Q04[1]', 0),
 ('id_2', 'G01Q04[2]', 1),
 ('id_1', 'G01Q04[2]', 1),
 ('id_4', 'G01Q04[2]', 0),
 ('id_2', 'G01Q04[3]', 0),
 ('id_1', 'G01Q04[3]', 0),
 ('id_4', 'G01Q04[3]', 1),
 ('id_2', 'G01Q05[1]', 1),
 

### 2.1. Inter-rator agreement on sentiments

In [24]:
atask_sentiment = agreement.AnnotationTask(data=data_sentiment)
print("Cohen's Kappa:", atask_sentiment.kappa())
print("Fleiss's Kappa:", atask_sentiment.multi_kappa())
print("Krippendorf's Alpha:", atask_sentiment.alpha())
print("Scott's Pi", atask_sentiment.pi())

Cohen's Kappa: 0.4186815278907559
Fleiss's Kappa: 0.418681527890756
Krippendorf's Alpha: 0.4242214082657433
Scott's Pi 0.41868152789075613


In [25]:
emotion1_res = emotion1.T
emotion1_res.columns = ['id_2', 'id_1','id_4']
emotion1_res = emotion1_res.drop('id')
emotion1_res

id_2  id_1  id_4
G01Q01[5].  __Previous_sentence:__Pero_ella,_co...     0     0     0
G01Q01[12].  __Previous_sentence:__Pero_ella,_c...     0     0     0
G01Q01[6].  __Previous_sentence:__Pero_ella,_co...     0     0     0
G01Q01[10].  __Previous_sentence:__Pero_ella,_c...     0     0     0
G01Q01[7].  __Previous_sentence:__Pero_ella,_co...     1     1     0
...                                                  ...   ...   ...
G00Q25[7]. Previous_sentence:_La_sal,_el_garava...     0     0     0
G00Q25[8]. Previous_sentence:_La_sal,_el_garava...     0     0     1
G00Q25[9]. Previous_sentence:_La_sal,_el_garava...     0     1     0
G00Q25[11]. Previous_sentence:_La_sal,_el_garav...     0     0     0
G00Q25[13]. Previous_sentence:_La_sal,_el_garav...     1     0     0

[1224 rows x 3 columns]

In [26]:
emotion2_res = emotion2.T
emotion2_res.columns = ['id_1', 'id_2','id_4']
emotion2_res = emotion2_res.drop('id')
emotion2_res

id_1  id_2  id_4
G00Q136[5]. Previous_sentence:__Diceles_palabra...     0     0     0
G00Q136[12]. Previous_sentence:__Diceles_palabr...     0     0     0
G00Q136[6]. Previous_sentence:__Diceles_palabra...     0     0     0
G00Q136[10]. Previous_sentence:__Diceles_palabr...     0     0     0
G00Q136[7]. Previous_sentence:__Diceles_palabra...     0     0     0
...                                                  ...   ...   ...
G00Q270[7]. Previous_sentence:__esto_quiere_dec...     0     0     0
G00Q270[8]. Previous_sentence:__esto_quiere_dec...     0     0     0
G00Q270[9]. Previous_sentence:__esto_quiere_dec...     0     0     0
G00Q270[11]. Previous_sentence:__esto_quiere_de...     0     0     0
G00Q270[13]. Previous_sentence:__esto_quiere_de...     0     1     0

[1216 rows x 3 columns]

In [32]:
emotion_res = pd.concat([emotion2_res], ignore_index=True)
emotion_res

id_1  id_2  id_4
0        0     0     0
1        0     0     0
2        0     0     0
3        0     0     0
4        0     0     0
...    ...   ...   ...
1211     0     0     0
1212     0     0     0
1213     0     0     0
1214     0     0     0
1215     0     1     0

[1216 rows x 3 columns]

In [33]:
data_emotion_res = []
for idx, row in emotion_res.iterrows():
    data_emotion_res.append(("id_2", idx, row["id_2"]))
    data_emotion_res.append(("id_1", idx, row["id_1"]))
    data_emotion_res.append(("id_4", idx, row["id_4"]))
data_emotion_res

[('id_2', 0, 0),
 ('id_1', 0, 0),
 ('id_4', 0, 0),
 ('id_2', 1, 0),
 ('id_1', 1, 0),
 ('id_4', 1, 0),
 ('id_2', 2, 0),
 ('id_1', 2, 0),
 ('id_4', 2, 0),
 ('id_2', 3, 0),
 ('id_1', 3, 0),
 ('id_4', 3, 0),
 ('id_2', 4, 0),
 ('id_1', 4, 0),
 ('id_4', 4, 0),
 ('id_2', 5, 0),
 ('id_1', 5, 0),
 ('id_4', 5, 1),
 ('id_2', 6, 1),
 ('id_1', 6, 0),
 ('id_4', 6, 0),
 ('id_2', 7, 0),
 ('id_1', 7, 0),
 ('id_4', 7, 0),
 ('id_2', 8, 0),
 ('id_1', 8, 1),
 ('id_4', 8, 0),
 ('id_2', 9, 0),
 ('id_1', 9, 0),
 ('id_4', 9, 0),
 ('id_2', 10, 0),
 ('id_1', 10, 0),
 ('id_4', 10, 0),
 ('id_2', 11, 0),
 ('id_1', 11, 0),
 ('id_4', 11, 0),
 ('id_2', 12, 0),
 ('id_1', 12, 0),
 ('id_4', 12, 0),
 ('id_2', 13, 0),
 ('id_1', 13, 0),
 ('id_4', 13, 0),
 ('id_2', 14, 0),
 ('id_1', 14, 0),
 ('id_4', 14, 0),
 ('id_2', 15, 0),
 ('id_1', 15, 0),
 ('id_4', 15, 1),
 ('id_2', 16, 1),
 ('id_1', 16, 0),
 ('id_4', 16, 0),
 ('id_2', 17, 0),
 ('id_1', 17, 0),
 ('id_4', 17, 0),
 ('id_2', 18, 0),
 ('id_1', 18, 1),
 ('id_4', 18, 0),
 ('i

### 2.2 Inter-rator agreement on emotions

In [34]:
atask_emotion = agreement.AnnotationTask(data=data_emotion_res)
print("Cohen's Kappa:", atask_emotion.kappa())
print("Fleiss's Kappa:", atask_emotion.multi_kappa())
print("Krippendorf's Alpha:", atask_emotion.alpha())
print("Scott's Pi", atask_emotion.pi())

Cohen's Kappa: 0.18004591099621245
Fleiss's Kappa: 0.1812044383806009
Krippendorf's Alpha: 0.1812439136236932
Scott's Pi 0.18101941236611807


### 2.3 Agreement both sentiments and emotions

In [205]:
atask_all = agreement.AnnotationTask(data=all_survey_res)
print("Cohen's Kappa:", atask_all.kappa())
print("Fleiss's Kappa:", atask_all.multi_kappa())
print("Krippendorf's Alpha:", atask_all.alpha())
print("Scott's Pi", atask_all.pi())

Cohen's Kappa: 0.31116197107885774
Fleiss's Kappa: 0.31113731256777977
Krippendorf's Alpha: 0.3249349066533709
Scott's Pi 0.3111199462306653


# 3. Study with four annotators (round 3)


In [40]:
four_annotators = pd.read_csv('input/3_round/4_annotators.csv', delimiter=',', encoding='utf-8')
four_annotators

id. Response ID an0tatorname. Please state you name:  \
0                1                            Elisabeth   
1                2                               Yvonne   
2                4                               Ertler   
3                5                     Pascal Striedner   

   G00Q136[1]. Previous sentence:  Diceles palabras injuriosas, y ellos la responden sin decoro.  To an0tate:   Miralos sin humanidad, y como esclavos, y ellos à su vez la sirven sin respeto, sublevados contra su tyranìa.   Following sentence:   Suspendese por un rato esta gresca, y passa Madama al Tocador.   [positive]  \
0                                                  0                                                                                                                                                                                                                                                                                
1                                                  0                                                                                                                                                                                                                                                                                
2                                                  0                                                                                                                                                                                                                                                                                
3                                                  0                                                                                                                                                                                                                                                                                

   G00Q136[2]. Previous sentence:  Diceles palabras injuriosas, y ellos la responden sin decoro.  To an0tate:   Miralos sin humanidad, y como esclavos, y ellos à su vez la sirven sin respeto, sublevados contra su tyranìa.   Following sentence:   Suspendese por un rato esta gresca, y passa Madama al Tocador.   [neutral]  \
0                                                  0                                                                                                                                                                                                                                                                               
1                                                  0                                                                                                                                                                                                                                                                               
2                                                  0                                                                                                                                                                                                                                                                               
3                                                  0                                                                                                                                                                                                                                                                               

   G00Q136[3]. Previous sentence:  Diceles palabras injuriosas, y ellos la responden sin decoro.  To an0tate:   Miralos sin humanidad, y como esclavos, y ellos à su vez la sirven sin respeto, sublevados contra su tyranìa.   Following sentence:   Suspendese por un rato esta gresca, y passa Madama al Tocador.   [negative]  \
0                                                  1                                                                                   

In [41]:
four_annotators.columns = [col.split('.')[0] for col in four_annotators.columns]


In [42]:
four_annotators

id      an0tatorname  G00Q136[1]  G00Q136[2]  G00Q136[3]  G00Q137[1]  \
0   1         Elisabeth           0           0           1           0   
1   2            Yvonne           0           0           1           0   
2   4            Ertler           0           0           1           0   
3   5  Pascal Striedner           0           0           1           0   

   G00Q137[2]  G00Q137[3]  G00Q138[1]  G00Q138[2]  ...  G00Q267[3]  \
0           0           1           0           1  ...           1   
1           0           1           0           1  ...           1   
2           0           1           0           0  ...           1   
3           0           1           0           1  ...           0   

   G00Q268[1]  G00Q268[2]  G00Q268[3]  G00Q269[1]  G00Q269[2]  G00Q269[3]  \
0           1           0           0           0           1           0   
1           1           0           0           0           1           0   
2           0           0           1           0           0           1   
3           1           0           0           0           1           0   

   G00Q270[1]  G00Q270[2]  G00Q270[3]  
0           1           0           0  
1           1           0           0  
2           1           0           0  
3           1           0           0  

[4 rows x 407 columns]

In [43]:
 four_annotators.drop('an0tatorname', axis=1,inplace=True)



In [44]:
four_annotators

id  G00Q136[1]  G00Q136[2]  G00Q136[3]  G00Q137[1]  G00Q137[2]  G00Q137[3]  \
0   1           0           0           1           0           0           1   
1   2           0           0           1           0           0           1   
2   4           0           0           1           0           0           1   
3   5           0           0           1           0           0           1   

   G00Q138[1]  G00Q138[2]  G00Q138[3]  ...  G00Q267[3]  G00Q268[1]  \
0           0           1           0  ...           1           1   
1           0           1           0  ...           1           1   
2           0           0           1  ...           1           0   
3           0           1           0  ...           0           1   

   G00Q268[2]  G00Q268[3]  G00Q269[1]  G00Q269[2]  G00Q269[3]  G00Q270[1]  \
0           0           0           0           1           0           1   
1           0           0           0           1           0           1   
2           0           1           0           0           1           1   
3           0           0           0           1           0           1   

   G00Q270[2]  G00Q270[3]  
0           0           0  
1           0           0  
2           0           0  
3           0           0  

[4 rows x 406 columns]

In [45]:
four_annotators_t = four_annotators.T
four_annotators_t.columns = ['id_1','id_2', 'id_3','id_4']
four_annotators_t = four_annotators_t.drop('id')
four_annotators_t

id_1  id_2  id_3  id_4
G00Q136[1]     0     0     0     0
G00Q136[2]     0     0     0     0
G00Q136[3]     1     1     1     1
G00Q137[1]     0     0     0     0
G00Q137[2]     0     0     0     0
...          ...   ...   ...   ...
G00Q269[2]     1     1     0     1
G00Q269[3]     0     0     1     0
G00Q270[1]     1     1     1     1
G00Q270[2]     0     0     0     0
G00Q270[3]     0     0     0     0

[405 rows x 4 columns]

In [46]:
four_annotators_list = []
for idx, row in four_annotators_t.iterrows():
        four_annotators_list.append(("id_1", idx, row["id_1"]))
        four_annotators_list.append(("id_2", idx, row["id_2"]))
        four_annotators_list.append(("id_3", idx, row["id_3"]))
        four_annotators_list.append(("id_4", idx, row["id_4"]))


In [47]:
four_annotators_list

[('id_1', 'G00Q136[1]', 0),
 ('id_2', 'G00Q136[1]', 0),
 ('id_3', 'G00Q136[1]', 0),
 ('id_4', 'G00Q136[1]', 0),
 ('id_1', 'G00Q136[2]', 0),
 ('id_2', 'G00Q136[2]', 0),
 ('id_3', 'G00Q136[2]', 0),
 ('id_4', 'G00Q136[2]', 0),
 ('id_1', 'G00Q136[3]', 1),
 ('id_2', 'G00Q136[3]', 1),
 ('id_3', 'G00Q136[3]', 1),
 ('id_4', 'G00Q136[3]', 1),
 ('id_1', 'G00Q137[1]', 0),
 ('id_2', 'G00Q137[1]', 0),
 ('id_3', 'G00Q137[1]', 0),
 ('id_4', 'G00Q137[1]', 0),
 ('id_1', 'G00Q137[2]', 0),
 ('id_2', 'G00Q137[2]', 0),
 ('id_3', 'G00Q137[2]', 0),
 ('id_4', 'G00Q137[2]', 0),
 ('id_1', 'G00Q137[3]', 1),
 ('id_2', 'G00Q137[3]', 1),
 ('id_3', 'G00Q137[3]', 1),
 ('id_4', 'G00Q137[3]', 1),
 ('id_1', 'G00Q138[1]', 0),
 ('id_2', 'G00Q138[1]', 0),
 ('id_3', 'G00Q138[1]', 0),
 ('id_4', 'G00Q138[1]', 0),
 ('id_1', 'G00Q138[2]', 1),
 ('id_2', 'G00Q138[2]', 1),
 ('id_3', 'G00Q138[2]', 0),
 ('id_4', 'G00Q138[2]', 1),
 ('id_1', 'G00Q138[3]', 0),
 ('id_2', 'G00Q138[3]', 0),
 ('id_3', 'G00Q138[3]', 1),
 ('id_4', 'G00Q138[3

In [48]:
atask_four_an = agreement.AnnotationTask(data=four_annotators_list)
print("Cohen's Kappa:", atask_four_an.kappa())
print("Fleiss's Kappa:", atask_four_an.multi_kappa())
print("Krippendorf's Alpha:", atask_four_an.alpha())
print("Scott's Pi", atask_four_an.pi())

Cohen's Kappa: 0.46296296296296285
Fleiss's Kappa: 0.462962962962963
Krippendorf's Alpha: 0.4632944673068129
Scott's Pi 0.46296296296296285
